### ISSUE TITLE
X-Endstop/Homing not working

### ISSUE THREAD

**Issue by gegiarmo** [MarlinFirmware/Marlin#2071]
I'm trying to setup my new Azteeg X3 Pro, printer was previously working well on RAMPS 1.4. As a note, it's a custom printer, but works as a standard Cartesian. I've downloaded the release version of Marlin, updated feedrates, endstop inversion, etc. as necessary.

The issue I'm running into is that it has weird X-endstop behavior. Y-axis is set up exactly the same and behaves as it should. When the X-endstop is not inverted, homing only makes the axis progress a bit, similar to what @emartinez167 ran into with Issue #1725. If I'm just moving the axis in the + direction, it works fine, but when trying to move in the - direction, even when not at X=0mm, it claims the endstop was hit. When the endstop is inverted, it completely ignores it, whether it is triggered or not.

Here is a pastebin of the configuration.h: [here](http://pastebin.com/4bB56pU1).

**Comment by thinkyhead** [comment]
@gegiarmo There have been enough changes in the configuration files that you probably need to doa refresh. If you have some tool that can do a diff between your `Configuration.h` and `Configuration_adv.h` files and the latest versions, that will help you to update them to be compliant with the latest code. Or, just start with the fresh files and then apply your customizations. After that, try again, and if the issue still exists we can look further.

**Comment by gegiarmo** [comment]
Started fresh with the latest development build. Still having the same behavior.

**Comment by thinkyhead** [comment]
@gegiarmo Do you see the same behavior with `ENDSTOPS_ONLY_FOR_HOMING` enabled and also disabled? (Trying to see if, or how, that setting might be involved.)

**Comment by gegiarmo** [comment]
Yes, same behavior with ENDSTOPS_ONLY_FOR_HOMING both enabled and disabled.

**Comment by gegiarmo** [comment]
So, fun story. Per the Azteeg X3 Pro pin definitions [here](http://files.panucatt.com/datasheets/x3pro_pins.pdf), the X-min is pin 2 and max is pin 3, the opposite of RAMPS and as defined in Marlin. So this is what I set it to, because I thought Marlin was wrong. The punch line: The Azteeg documentation is wrong, pin 3 is correct for X-min. It works now. Thanks for the input though @thinkyhead .

**Comment by thinkyhead** [comment]
I'm letting them know...

> Panucatt, I see that the PDF you have posted at http://files.panucatt.com/datasheets/x3pro_pins.pdf has an error, which is causing some users to become confused. The PDF shows X MIN ENDSTOP as Pin 2 and X MAX ENDSTOP as Pin 3. This is incorrect, and is not aligned with RAMPS. The pin numbers should be reversed, with X MIN ENDSTOP as Pin 3 and X MAX ENDSTOP as Pin 2. Please correct this and post a new PDF.

**Comment by gegiarmo** [comment]
Yup, I already did the same, hopefully they do update it.

**Comment by dvseliteus** [comment]
I am having the same exact issue with my Azteeg X3 and the X-Axis End stop not working.
I have changes the pin assignment on the pin.h tab in the Marlin firmware and when I press the limit switch nothing happens?

``` cpp
#if MOTHERBOARD == 99
#define KNOWN_BOARD 1

#define X_STEP_PIN          3
#define X_DIR_PIN           2
#define X_ENABLE_PIN        -1
#define X_STOP_PIN          16
```

I use simplify3d and when I hit the home button it triggers X to zero out even though it has not hit the switch?
Anyone have an idea what's wrong here?

**Comment by thinkyhead** [comment]
@dvseliteus What version of Marlin are you using?

**Comment by dvseliteus** [comment]
Hi @thinkyhead thank you for the quick reply.
I'm using Marlin-1.0.2 here are my [pin.h](http://pastebin.com/CGYfbg7w) and [configuration.h](http://pastebin.com/suMwxPD5) files.
I reached out to panucatt for support but have not gotten any reply. I'm stuck at this point.
Any help would be greatly appreciated. Thank you!

**Comment by thinkyhead** [comment]
@dvseliteus

With the line "`#if MOTHERBOARD == 99`" it looks like an older version of Marlin. I recommend trying 1.0.2-1 or 1.1.0-RC7 and see if it works better for you.

The Azteeg X3 has most pins identical to RAMPS 1.4 (Extruder, Fan, Bed). The pins file with 1.0.2-1 seems correct. Anyway, here are the proper pins for the ones you mentioned above:

``` cpp
#define X_STEP_PIN         54
#define X_DIR_PIN          55
#define X_ENABLE_PIN       38
#define X_MIN_PIN           3
#define X_MAX_PIN           2
```

**Comment by dvseliteus** [comment]
Thanks again for your help @thinkyhead 
I have the motherboard set as 67?
I have entered the pin numbners under the #67 pins and I am still having the same issue.
I am hesitant to try 1.1.0-RC7 as I will have to re-program everything and I don't really know what I am doing. Of all the things I thought would get me stumped I can;t believe it's something like this!

**Comment by dvseliteus** [comment]
Here is what panucatt has provided for support;
"it is possible that the endstop pin on the Atmel chip is not working anymore. It can be caused by static getting into the pin from the endstop wiring, Ive seen a few of these happen on X3s and X3 Pros. What you can do is to use the MAX endstop pins for the X endstop by defining the X endstop using the pin of the MAX endstop. Say if you will be using the Xmax as your Xmin endstop you can use D2 instead of D3 which is what the Xmin uses. Actually you can use any free pin and just make sure its pulled up in firmware."

Now here is where I get confused. Where in the firmware exactly am I making these changes in correspondence with the pins on the board? I just can't seem to wrap my head around this. Can someone please dumb it down for me?
[http://pastebin.com/CGYfbg7w](url) [http://pastebin.com/suMwxPD5](url)

I have tried all 3 pins for the X axis on the board it must be something in the firmware. I'd like to stick to Marlin 1.0.2-1 I tried working with 1.1.0-RC7 and I'm getting weird compile errors like "Error compiling for board Arduino/Genuino Mega or Mega 2560." I feel like I'm just going in endless circles of frustration.

**Comment by thinkyhead** [comment]
> I am hesitant to try 1.1.0-RC7 as I will have to re-program everything

You'll be better off using `RCBugFix` or `1.1.0-RC7` as the newer Marlin places the pins for each board in its own file and fixes many bugs. Anyway, to swap the X endstop pins for your board in `1.0.2-1` all you need to do is change [lines 599-600 in `pins.h`](https://github.com/MarlinFirmware/Marlin/blob/1.0.x/Marlin/pins.h#L599-L600):

``` diff
-   #define X_MIN_PIN           3
-   #define X_MAX_PIN           2
+   #define X_MIN_PIN           2
+   #define X_MAX_PIN           3

```

**Comment by dvseliteus** [comment]
Thanks again @thinkyhead I have 1.1.0-RC7 all setup but I am getting a strange "Error compiling for board Arduino/Genuino Mega or Mega 2560." and I'm not able to upload to the board.

I'll try the fix above on 1.0.2-1 and see if I can at least get the x-stop triggering and then focus back to the newer version.

**Comment by thinkyhead** [comment]
@dvseliteus Which version of Arduino IDE are you using?

**Comment by dvseliteus** [comment]
@thinkyhead arduino-1.6.10, I need to use 1.6.0 or later (earlier) don't I...

**Comment by thinkyhead** [comment]
Version 1.6.10 seems to be acting up. Try 1.6.9 or 1.6.8.

**Comment by dvseliteus** [comment]
Ok you got me working with the 1.1.0-RC7 version and I quite like the format over 1.0.2-1. I have tried Arduino version 1.6.10, 1.6.0, installed U8glib library, 1.6.8, and finally 1.6.9.
I keep getting a compile error; 
SanityCheck.cpp:554: error: #error "HEATER_1_PIN not defined for this board."
     #error "HEATER_1_PIN not defined for this board."
I looked at your thread https://github.com/MarlinFirmware/Marlin/issues/1741 but couldn't figure it out.
[http://pastebin.com/BMk2qkff](url) Line 554

**Comment by Blue-Marlin** [comment]
Please append the Configuration.h causing the error.

**Comment by dvseliteus** [comment]
@Blue-Marlin http://pastebin.com/U2ffANSJ here you go :+1:

**Comment by Blue-Marlin** [comment]
pins_AZTEEG_X3.h was not prepared for two extruders. Please try the folowing patch.
Feedback please.

``` diff
@@ -24,11 +24,18 @@
  * AZTEEG_X3 Arduino Mega with RAMPS v1.4 pin assignments
  */

 #define BOARD_NAME "Azteeg X3"

-#include "pins_RAMPS_14_EFB.h"
+#if EXTRUDERS == 1
+  #define IS_RAMPS_EFB
+#endif
+#if EXTRUDERS == 2
+  #define IS_RAMPS_EEB
+#endif
+
+#include "pins_RAMPS_14.h"

 #if ENABLED(VIKI2) || ENABLED(miniVIKI)

   #undef DOGLCD_A0
   #undef DOGLCD_CS
```

**Comment by thinkyhead** [comment]
Is 2 heaters the maximum for Azteeg X3? If so, I'll add a check in its pins file for `HOTENDS > 2`.

**Comment by Blue-Marlin** [comment]
Yes.
http://reprap.org/mediawiki/images/a/a5/X3_wiring_diagram.jpg

**Comment by thinkyhead** [comment]
Feel free to just post a link to http://LMGTFY.com for these kinds of questions. 😜

**Comment by Blue-Marlin** [comment]
All RAMPS combinations could work. But because of the extra low power mosfets there are enough extra connectors for fans. So these two make most sense.

**Comment by dvseliteus** [comment]
@Blue-Marlin thank you for pointing this out.
I went ahead and added the code to the pins_AZTEEG_X3.h tab but I am receiving the following error;
[http://pastebin.com/4JY1a6KY](url)
Here is my revised pins_AZTEEG_X3.h file: http://pastebin.com/pgHi0czA
Feels like we are getting close here, thanks for the support all!

**Comment by Blue-Marlin** [comment]
@dvseliteus 
Applying a diff means removing the lines beginning with '-' and adding that ones, beginning with '+' - without the '+'.
Don't worry. The changes are already in RCBugFix.

**Comment by dvseliteus** [comment]
@Blue-Marlin 
Thanks, I have downloaded the RCBugFix and tried my best to make the modifications needed to run the Azteeg X3 and Viki LCD (I'm a begginer)
I am running into an error: #error "HEATER_1_PIN not defined for this board."
I tired to implement some fixes from this thread: [https://github.com/MarlinFirmware/Marlin/issues/3497](url) but was unsuccessful.
Here is a download link to my Marlin Folder: <a href=http://www.filedropper.com/marlin110-rcbugfix-26july2016><img src=http://www.filedropper.com/download_button.png width=40 height=50 border=0/></a><br /><div style=font-size:9px;font-family:Arial, Helvetica, sans-serif;width:127px;font-color:#44a854;></a></div>
I'm not sure how to properly set the Z_MIN_PROBE correctly?

**Comment by thinkyhead** [comment]
@dvseliteus I just submitted a patch to `RCBugFix` that should fix your `HEATER_1_PIN` error. It adds code to assign the power outputs to different uses for RAMPS derivative boards, which were not being auto-selected previously. On RAMPS the pin labeled D10 is for the first extruder, D9 is for a fan or a second extruder and D8 is used for the heated bed or a second fan (so you have the options of EEF, EEB, EFF, EFB as with RAMPS). The same connection order will apply on your Azteeg X3, although the plug labels may differ.

**Comment by dvseliteus** [comment]
Thanks @thinkyhead  I will continue to keep at it and see if I can get the code to eventually compile.
I am able to get the firmware to compile with 1 Extruder.
2 Extruders I am getting #error "HEATER_1_PIN not defined for this board." no matter what I configure the pins_RAMPS.h pins. (Waiting for submitted a patch to RCBugFix that should fix your HEATER_1_PIN error.)

When I compile with 1 extruder I am getting a weird error with the #define LCD_I2C_VIKI Panucatt VIKI LCD with status LEDs
'DIOBEEPER_PIN_RPORT' was not declared in this scope

I am going to buy a Ramps 1.4 Kit and also try hooking that up. That compiles with 2 extruders and #define ULTRA_LCD with no issues. I still want to use the Azteeg but need to push for a quick temp solution.

@thinkyhead @Blue-Marlin Thank you for your great support and troubleshooting. Hopefully we can get this fixed and post it for others if they want to use this hardware.

**Comment by thinkyhead** [comment]
> When I compile with 1 extruder I am getting a weird error with the #define LCD_I2C_VIKI Panucatt VIKI LCD with status LEDs 'DIOBEEPER_PIN_RPORT' was not declared in this scope

#4506 should fix this.

**Comment by dvseliteus** [comment]
[#4506](https://github.com/MarlinFirmware/Marlin/pull/4506) did the trick! Thanks @thinkyhead

**Comment by dvseliteus** [comment]
Viki menu does not work when button is pressed and the X Y and Z fields blink question marks?

**Comment by thinkyhead** [comment]
Blinking question marks is normal behavior. When you home an axis, that axis will stop blinking.

**Comment by dvseliteus** [comment]
Ok great that makes sense. I still have the original issue that first brought me to this forum X-Endstop/Homing not working #2071 with the RCBugFix. I will have to try to see what's going on here again as I cannot home X properly.

**Comment by Blue-Marlin** [comment]
Blinking
'X' <-> '?' means axis not homed.
'X' <-> ' ' means stepper has been shut down since last homing. If you don't like it turn it off via `#define DISABLE_REDUCED_ACCURACY_WARNING`

**Comment by thinkyhead** [comment]
I believe the PR #4506 should fix the encoder button issue. It will be merged shortly.

**Comment by dvseliteus** [comment]
@Blue-Marlin thank you for the info I quite like it actually now that I know what it means. Thanks @thinkyhead I got the beeper working but now when I press the button it shows infoscreen text on the first line for a half second and then reverts back to the main screen display.

**Comment by thinkyhead** [comment]
Test after #4506 is merged and see if it helps.

**Comment by Blue-Marlin** [comment]
Basic hardware endstop setup:
- Send M119 in mid air and  observe the output.
- No endstop should be triggered.
- If one shows as triggered try to switch it by hand. If it alters the state revert it (*_ENDSTOP_INVERTING true)
- If it does not trigger, alter the related `ENDSTOPPULLUP` and retest.
- Disable unused endstops (`//#define USE_XMAX_PLUG`)

Probe:
If you have a unused z-min-pin, use that.
If you have a z-min-endstop in use, define 'Z_MIN_PROBE_PIN' with the number of an unused endstop. Also define `Z_MIN_PROBE_ENDSTOP` and undefine `Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN`.

Always retest with M119 if all makes sense now.

**Comment by dvseliteus** [comment]
@thinkyhead I am having the same result after applying the updated code to the 6 tabs on #4506 fix.
Here is what it looks like when I try to access the menu: https://youtu.be/mE4ppKlIf-U
@Blue-Marlin I am using simplify 3D to send an M119 command and nothing happens, I am not sure how to proceed next?
Thanks again for sticking with me on these issues.

**Comment by Blue-Marlin** [comment]
@dvseliteus 
Don't use non free software.
I'm quite sure there is an option to make Marlin's output visible in the log.

**Comment by dvseliteus** [comment]
@Blue-Marlin I have purchased Simplfy 3D and use it with my prusa printer.
I can copy a log file from the software if you like or I can test with a software you recommend?

**Comment by Blue-Marlin** [comment]
There is an option to make Marlin's output visible. I suppose that is off by default.

The output should look like 

```
23:55:44.754 : N5129 M119*21
23:55:44.759 : Reporting endstop status
23:55:44.759 : x_min: open
23:55:44.759 : y_min: open
23:55:44.759 : z_min: open
23:55:44.759 : ok
```

> Don't use non free software.

That's more a political statement. Regardless if you paid for it or not.

**Comment by thinkyhead** [comment]
> Don't use non free software

(By "non free" he doesn't mean costing no $¢£ money. He means "non free" or "non libre" in the sense of not coming with the freedoms given to you by the GPL, such as having the source code.)

**Comment by thinkyhead** [comment]
> Here is what it looks like when I try to access the menu

Very strange. It's as though the button is showing up as pressed when you press it. and then showing up again as pressed when you release it. What happens if you simply press-and-hold the button? Does it act as though it is being pressed repeatedly?

**Comment by dvseliteus** [comment]
@thinkyhead Yes when I  press-and-hold the button it just acts as it's pressed re-repeatedly. I'm starting to think as a beginner I picked the most difficult parts and their hardware versions to make work.
@thinkyhead @Blue-Marlin Thanks for that info I now understand it was in reference to an open source software. I am still unsure how to implement blue-marlins instructions.

**Comment by thinkyhead** [comment]
@dvseliteus Do you have the "`SPEAKER`" setting enabled? I'm wondering if the "debounce" that we use for the controller button is working properly with `SPEAKER` both enabled and disabled.

**Comment by dvseliteus** [comment]
@thinkyhead

``` cpp
if (this->state.tone.frequency > 0) {
  #if ENABLED(SPEAKER)
    ::tone(BEEPER_PIN, this->state.tone.frequency, this->state.tone.duration);
  #else
    this->on();
  #endif
```

Is there where we see if it is enabled on the buzzer.h file?

**Comment by thinkyhead** [comment]
In `Configuration.h` you will see:

``` cpp
//
// SPEAKER/BUZZER
//
// If you have a speaker that can produce tones, enable it here.
// By default Marlin assumes you have a buzzer with a fixed frequency.
//
//#define SPEAKER
```

**Comment by dvseliteus** [comment]
Speaker appears to be disabled.

**Comment by thinkyhead** [comment]
> Speaker appears to be disabled

Fabulous. Try enabling `SPEAKER` to see if it makes any difference. Then we will take a look at the `lcd_quick_feedback` function to make sure it has an adequate delay after playing the tone.

**Comment by dvseliteus** [comment]
Okay. Enabling `SPEAKER` yielded the same issue with the menu button.

**Comment by thinkyhead** [comment]
@dvseliteus On most controllers the controller button triggers once, then we have a delay and some logic to "debounce" the button, so it won't trigger again right away. Possibly the current method isn't reliable for your Viki. So, I will see about making a more robust button handler for you to test soon.

**Comment by dvseliteus** [comment]
Thank you @thinkyhead the Viki worked great in the older version of Marlin 1.0.2-1

Marlin 1.0.2-1 - X-Axis had weird X-endstop behavior but viki worked
Marlin RCBugFix - Viki button issue and same weird X-endstop behavior (won't trigger)

**Comment by thinkyhead** [comment]
Have you updated your configuration since http://pastebin.com/U2ffANSJ? I'll have a look at that now.

**Comment by thinkyhead** [comment]
@dvseliteus Use the Pronterface app instead of Simplify3D. Use `M119` from the console to test the endstops and see how they are behaving.

**Comment by dvseliteus** [comment]
@thinkyhead that last pastebin was of 1.1.0-RC7 version.
Here is the current version http://pastebin.com/rTb5e6YT RCBugFix

**Comment by thinkyhead** [comment]
Your configuration seems fine. How does the output from `M119` look?

**Comment by dvseliteus** [comment]
@thinkyhead here is the log when I send the M119

```
>>> M119
SENDING:M119
Reporting endstop status
x_min: open
y_min: open
z_min: open
```

**Comment by dvseliteus** [comment]
Here are the pending issues so far
1. When I hit the home X button the carriage judders and does not move continuously till triggering the X-End Stop. Here you can see this judder: https://youtu.be/Rrq3OccXRes each time the carriage moves I am hitting  the Home X button. Once the Carriage does hit the endstop I do get a proper trigger.
2. Now my Z-Probe does not trigger the bed to stop? I must have a wrong setting somewhere....
   http://pastebin.com/Jwu0HXUU <- Current Config.h
3. Can't access menu by pressing button https://youtu.be/mE4ppKlIf-U

**Comment by Blue-Marlin** [comment]
Your x-speed/acceleration is much higher than your machine can handle.

**Comment by dvseliteus** [comment]
@Blue-Marlin I have decreased the speed settings and still run into the same problem Homing the X Axis. It just happens much slower now, but is the same exact problem.

Here is the setting I tried.

``` cpp
// MOVEMENT SETTINGS
// @section motion
//

// default settings

#define DEFAULT_AXIS_STEPS_PER_UNIT   {80,80,4000,500}  // default steps per unit for Ultimaker
#define DEFAULT_MAX_FEEDRATE          {500, 500, 2, 45}    // (mm/sec)
#define DEFAULT_MAX_ACCELERATION      {450,450,100,10000}    // X, Y, Z, E maximum start speed for accelerated moves. E default values are good for Skeinforge 40+, for older versions raise them a lot.

#define DEFAULT_ACCELERATION          450    // X, Y, Z and E acceleration in mm/s^2 for printing moves
#define DEFAULT_RETRACT_ACCELERATION  450    // E acceleration in mm/s^2 for retracts
#define DEFAULT_TRAVEL_ACCELERATION   450    // X, Y, Z acceleration in mm/s^2 for travel (non printing) moves

// The speed change that does not require acceleration (i.e. the software might assume it can be done instantaneously)
#define DEFAULT_XYJERK                10.0    // (mm/sec)
#define DEFAULT_ZJERK                 0.4     // (mm/sec)
#define DEFAULT_EJERK                 5.0    // (mm/sec)
```

My Machine should handle 3000 with no issues?

**Comment by Blue-Marlin** [comment]
> #define DEFAULT_MAX_FEEDRATE {500, 500, 2, 45} // (mm/sec)

This is for x&y about twice as much as i'd expect to work.
For a similar machine i have

``` cpp
// Homing speeds (mm/m)
#define HOMING_FEEDRATE_XY (80*60)
#define HOMING_FEEDRATE_Z  (3.5*60)

//
// MOVEMENT SETTINGS
// @section motion
//

// default settings

#define DEFAULT_AXIS_STEPS_PER_UNIT   {200.0*16.0/2.0/17.0, 200.0*16.0/2.0/17.0, 200.0*16.0/0.8, (200.0*16.0*45.0/11.0/3.141592654)/7.2722}  // Achatz Sainsmart Wade
#define DEFAULT_MAX_FEEDRATE          {250, 360, 3.5, 20}    // (mm/sec)
#define DEFAULT_MAX_ACCELERATION      {4000,5000,200,5000}    // X, Y, Z, E maximum start speed for accelerated moves. E default values are good for Skeinforge 40+, for older versions raise them a lot.

#define DEFAULT_ACCELERATION          5000    // X, Y, Z and E acceleration in mm/s^2 for printing moves
#define DEFAULT_RETRACT_ACCELERATION  5000    // E acceleration in mm/s^2 for retracts
#define DEFAULT_TRAVEL_ACCELERATION   5000    // X, Y, Z acceleration in mm/s^2 for travel (non printing) moves

// The speed change that does not require acceleration (i.e. the software might assume it can be done instantaneously)
#define DEFAULT_XYJERK                15.0    // (mm/sec)
#define DEFAULT_ZJERK                 0.2     // (mm/sec)
#define DEFAULT_EJERK                 6.0    // (mm/sec)
```

If you have EEROM activated be sure to 
M502
M500

**Comment by dvseliteus** [comment]
Thanks @Blue-Marlin I will try this on Monday and let you know. Have a great weekend!

**Comment by thinkyhead** [comment]
> here is the log when I send the M119

Cool. So, does it always print out the same log, even when you hold down each of the endstops?

**Comment by dvseliteus** [comment]
@thinkyhead hope your weekend was well. 
Here are the readings sending M119 with a manual trigger on each axis.
X-Axis Manual Close

```
SENDING:M119
Reporting endstop status
x_min: open
y_min: open
z_min: open
```

Y-Axis Manual Close

```
SENDING:M119
Reporting endstop status
x_min: open
y_min: TRIGGERED
z_min: open
```

Z-Axis Manual Close (Probe)

```
SENDING:M119
Reporting endstop status
x_min: open
y_min: open
z_min: open
```

The X-Axis appears to be working correctly. The Z-Probe is not working but the red light comes on when triggered. I must have something wrong in the firmware.

My main concern though is why the Carriage won't home properly it must be the board right? So I guess the next step here would be to try "What you can do is to use the MAX endstop pins for the X endstop by defining the X endstop using the pin of the MAX endstop. Say if you will be using the Xmax as your Xmin endstop you can use D2 instead of D3 which is what the Xmin uses. Actually you can use any free pin and just make sure its pulled up in firmware."

I'm guessing I will look to the RAMPS.h file to make this change?

```
#define X_MIN_PIN           3
#ifndef X_MAX_PIN
  #define X_MAX_PIN         2
#endif
```

and just switch the 3 & 2 pins and rewire the endstop on the board.

**Comment by dvseliteus** [comment]
Okay I have successfully used the Xmax pins as my Xmin tested the M119 to make sure thr tigger works and it does. I still receive the same jitter motion when I try to home the X-Axis. The Y-Axis homes with no problems.
So now I need to know do you think it is the board or is it something in the firmware that is causing this to happen?

**Comment by thinkyhead** [comment]
Under normal circumstances, and with reasonable speeds configured, you should not see any "jitter" when moving an axis. Using your host, try moving X around at different speeds, starting from your "homing speed" of 80mm/s (4800mm/m), and see if any jitter appears. Then try some lower speeds and see how it behaves.

**Comment by dvseliteus** [comment]
@thinkyhead Speeds 4800, 3000, 2000, 1000, 500, 200, 100, 50 and 25mm/min all yield the same result.
I tried a different Nema motor as well and it behaves the same. Why would the Y-Axis work fine at all the speeds but not the X-Axis? This issue must be with board right?

@thinkyhead @Blue-Marlin Can you recommend the best board and LCD for this printer and Marlin Firmware?

**Comment by dvseliteus** [comment]
I'm thinking RAMBo v1.3 might be a good board swap.

**Comment by thinkyhead** [comment]
> Why would the Y-Axis work fine at all the speeds but not the X-Axis?

That certainly sounds like hardware. I presume you have already tuned the current on the stepper drivers, so all the motors are getting adequate power, but not too much?

**Comment by dvseliteus** [comment]
@thinkyhead the stepper drivers have been tuned and I also tried swapping the drivers, then swapping pins from x/y axis motors, then switching motors and finally quitting.
I ordered a RAMBo v1.3 board and have already setup the firmware so hopefully it will get going "out of the box" since I can tune the stepper driver though the firmware...
The only downside is I will now have to convert all of the connectors from male jumper to female dupont.
Thanks for your and @Blue-Marlin s help. I have learned a lot and really like the Marlin Firmware over the other ones out there.

**Comment by dvseliteus** [comment]
@thinkyhead 
I did an experiment and I changed the X/Y wiring on the board but left the pin assignments the same on the ramps pins. Strange thing is when I home the Y-Axis which is actually the X-Axis I get the same issue. When i home the X-Axis now moving in Y Direction is homes with no issues?

What could be the possible causes here?
1. Stepper Motor is Bad? (Swapped motors and the same thing happened)
2. Firmware issue?
3. Board (how could it be the board if this result happened?)

This is really weird...

**Comment by thinkyhead** [comment]
@dvseliteus What if you swap both the connectors _and the pin assignments_? Does the X axis still act up then, or does the problem shift to the Y axis? Does your board have stepper drivers that you can move to other slots, or are they soldered onto the board? If you can swap them, then try that.

**Comment by dvseliteus** [comment]
@thinkyhead It so weird when I swap pins and wires I still get the same error on the x-axis.
I have switched stepper drivers and get the same error. I can't seem to replicate the problem on any other axis no mater what i try. I have checked and the carriage moves freely without and friction.

**Comment by thinkyhead** [comment]
@dvseliteus Trés bizarre. So the only thing common over all these tests is the stepper motor itself. So you might want to move that stepper motor to your Z axis, or just throw it in the trash. Or, try a ferrite bead on the wires.

**Comment by dvseliteus** [comment]
@thinkyhead right I have also changed to a typical NEMA 17 Stepper Motor and it also acted that way.

The true test will be when I install the RAMBo 1.3

**Comment by thinkyhead** [comment]
Indeed. I can't think of any reason, outside of having improper settings, that only one axis would act this way.

**Comment by dvseliteus** [comment]
@thinkyhead ok now this is interesting.
When I disconnect the endstop from the board pins and press the home X command the carriage moves with no problems. It's only when I have the endstop connected to the board that this weird jitter action happens? I tried different end stop and received the same results.

Weirder thing when I turn on the printer and check the stepper voltage it reads 0.7v (good) when i home the x axis and then measure the voltage it reads -4.10v?

**Comment by thinkyhead** [comment]
You could try disabling `ENDSTOPPULLUPS` and see if that makes any difference. But this is starting to sound like something is shorted on your board. Have you tried other endstop switches, or plugging into a different endstop plug (e.g., X_MAX) and swapping the pin definitions? Keep trying things to isolate whether it's the plug, the switch, or something else.

**Comment by dvseliteus** [comment]
@thinkyhead 
I tried disabling `ENDSTOPPULLUPS` but there was no difference.
I switched the wires to the Y-Axis Xmax pins and then changed the pin assignment on the ramps board to `#define X_MIN_PIN 15` which also had the same result.
It seems no matter what I do this X-Axis just will not home without skipping.
The RAMBo 1.3 should arrive today so hopefully this week I will have some better news.
